editing/fx/passes/pact/approximate.py

```diff
@@ -57,28 +51,25 @@ class ApproximateSoftmaxPass(SequentialPass):
 
     modes = ["I-BERT", "ITA", 'ITA-Partial']
 
-    def __init__(self, mode: Literal["I-BERT", "ITA", 'ITA-Partial'] = "I-BERT", **kwargs):
+    def __init__(self, symbolic_trace: callable = PACT_symbolic_trace, mode: Literal["I-BERT", "ITA", 'ITA-Partial'] = "I-BERT", **kwargs):
         passes = []
         pattern = nn.Sequential(nn.Softmax())
-
         assert mode in self.modes, f"[ApproximateSoftmaxPass] Invalid mode {mode} specified!"
-
-        passes.append(ReplaceSequentialPatternPass(pattern, PACT_symbolic_trace, partial(replSoftmax, mode=mode), f'_APPROXIMATE_SOFTMAX_PASS'))
-
+        passes.append(ReplaceSequentialPatternPass(pattern, symbolic_trace, partial(replSoftmax, mode=mode), f'_APPROXIMATE_SOFTMAX_PASS'))
         super().__init__(*passes, name_prefix='_APPROXIMATE_SOFTMAX_PASS')
 
 class ApproximateGELUPass(SequentialPass):
-    def __init__(self, **kwargs):
+    def __init__(self, symbolic_trace: callable = PACT_symbolic_trace, **kwargs):
         passes = []
         pattern = nn.Sequential(nn.GELU())
-        passes.append(ReplaceSequentialPatternPass(pattern, PACT_symbolic_trace, lambda x,y: PACTGELU(), f'_APPROXIMATE_GELU_PASS'))
+        passes.append(ReplaceSequentialPatternPass(pattern, symbolic_trace, lambda x,y: PACTGELU(), f'_APPROXIMATE_GELU_PASS'))
         super().__init__(*passes, name_prefix='_APPROXIMATE_GELU_PASS')
 
-class ApproximateSiLUWithGELUPass(SequentialPass):
-    def __init__(self, custom_trace, **kwargs):
+class ApproximateSiLUPass(SequentialPass):
+    def __init__(self, symbolic_trace: callable = PACT_symbolic_trace, **kwargs):
         passes = []
         pattern = nn.Sequential(nn.SiLU())
-        passes.append(ReplaceSequentialPatternPass(pattern, custom_trace, lambda x,y: PACTGELU(), f'_APPROXIMATE_SILU_PASS'))
+        passes.append(ReplaceSequentialPatternPass(pattern, symbolic_trace, lambda x,y: PACTHardswish(eps_s=0.005), f'_APPROXIMATE_SILU_PASS'))
```

Comment:
Why is eps_s = 0.005?

Reply:
Sorry, no. Look at the function and understand what eps_s is used for. If it used for constants only, it would not be a parameter!

Reply:
`one_over_six = PACTQuantize(o/6, self.eps_s, 0., 1., floor=False)`
self.eps_s is used to requantize o/6 in the range of 0...1. In this case eps_s should be `1/n_levels`.

Reply:
`n_levels`, in this case, we can probably pick to be well-representable in standard hardware, e.g. 2**8-1, 2**16-1, or something like that.

Reply:
Addressed in [this commit](https://github.com/pulp-platform/quantlib/pull/15/commits/c4eba483ac80387ce0f91c4d14d297ac2d90078b). I modified quantizedLlama's `quant.py` script at [this commit](https://iis-git.ee.ethz.ch/victor.jung/quantizedllama/-/commit/4c286a54ead886d06d2a6415543ec037d510892d) accordingly.